test/spec/grammar/dsl_api_test.js

```diff
@@ -0,0 +1,140 @@
+const expect = require('chai').expect;
+const tokens = require('../../../lib/dsl/lexer').tokens;
+const getSyntacticAutoCompleteSuggestions = require('../../../lib/dsl/api')
+  .getSyntacticAutoCompleteSuggestions;
+const parse = require('../../../lib/dsl/api')
+  .parse;
+
+
+describe('JDL DSL API', () => {
+  describe('when wanting an AST', () => {
+    describe('with a valid input', () => {
+      it('returns an AST', () => {
+        const ast = parse(`
+          entity A {
+            field String,
+          }`);
+        expect(ast.entities).to.have.lengthOf(1);
+        expect(ast.entities[0]).to.deep.eql({
+          name: 'A',
+          tableName: 'A',
+          body: [
+            {
+              name: 'field',
+              type: 'String',
+              validations: [],
+              javadoc: null
+            }
+          ],
+          javadoc: null
+        });
+      });
+    });
+
+    describe('with a lexing error', () => {
+      it('throws an error', () => {
+        const invalidInput = 'entity ± {';
+        expect(() => parse(invalidInput)).to
+          .throw('unexpected character: ->±<- at offset: 7, skipped 1 characters.');
```

Comment:
If you want to test that I suggest having more than one `it`:
  - `throws an error` => to test there's an error
  - `displays the illegal character` => to test the illegal character is thrown back at us (which is **nice** by the way)
  - `displays column and line information` => to test there's some kind of info about that

If you decide to change the API for whatever reason, I want my test to fail and yell at me that I should read the doc/changelog to be up-to-date.

Reply:
I just want to see that it throws "something" I will make it more general.